tests/rptest/clients/types.py

```diff
@@ -83,3 +83,21 @@ def __eq__(self, other):
     def _random_topic_suffix(self, size=10):
         return "".join(
             random.choice(string.ascii_lowercase) for _ in range(size))
+
+    def extra_config(self):
+        conf = {
+            'cleanup_policy': self.cleanup_policy,
+            'compression_type': self.compression_type,
+            'message.timestamp.type': self.message_timestamp_type,
+            'segment.bytes': self.segment_bytes,
+        }
+        if self.retention_bytes:
```

Comment:
I can't remember if zero is valid for retention_bytes and retention_ms, but to be on the safe side I would tend to use an "is not None" for these conditions, rather than truthiness checks that might trip up on zeros.